src/test/e2e/dev_test.go

```diff
@@ -145,13 +148,35 @@ func TestDevDeploy(t *testing.T) {
 	t.Run("Test dev deploy with --set flag", func(t *testing.T) {
 		bundleDir := "src/test/bundles/02-variables"
 		bundleTarballPath := filepath.Join(bundleDir, fmt.Sprintf("uds-bundle-variables-%s-0.0.1.tar.zst", e2e.Arch))
-		_, stderr := runCmd(t, "dev deploy "+bundleDir+" --set ANIMAL=Longhorns --set COUNTRY=Texas -l=debug")
+		_, stderr := runUDSCmd(t, "dev deploy "+bundleDir+" --set ANIMAL=Longhorns --set COUNTRY=Texas -l=debug")
 		require.Contains(t, stderr, "This fun-fact was imported: Longhorns are the national animal of Texas")
 		require.NotContains(t, stderr, "This fun-fact was imported: Unicorns are the national animal of Scotland")
 		remove(t, bundleTarballPath)
 	})
 
-	// delete packages because other tests depend on them being created with SBOMs (ie. force other tests to re-create)
-	e2e.DeleteZarfPkg(t, "src/test/packages/podinfo")
-	e2e.DeleteZarfPkg(t, "src/test/packages/nginx")
+	t.Run("Test dev deploy with excluded components", func(t *testing.T) {
+		// remove any lingering podinfo and nginx deployments
+		e2e.UDS(strings.Split("zarf tools kubectl delete deploy -n podinfo podinfo", " ")...)
+		e2e.UDS(strings.Split("zarf tools kubectl delete deploy -n nginx nginx-deployment", " ")...)
+
+		bundleDir := "src/test/bundles/15-dev-deploy/optional"
+		bundleName := "dev-deploy-exclude"
+		bundleTarballPath := filepath.Join(bundleDir, fmt.Sprintf("uds-bundle-%s-%s-0.0.1.tar.zst", bundleName, e2e.Arch))
+		// exclude nginx from podinfo-nginx pkg
+		runUDSCmd(t, "dev deploy "+bundleDir+" --exclude-components=podinfo-nginx.nginx")
+
+		// inspect created bundle and ensure nginx isn't included
+		// todo: refactor when we move 'inspect' to stdout and add --no-color flag
```

Comment:
cc @TristanHoladay , even when we move the output of `inspect` to `stdout` we still run into color issues. We'll also need to add a `--no-color` flag like Zarf does